docs/kubernetes/kctlr-rewrite-url.rst

```diff
@@ -0,0 +1,221 @@
+:product: BIG-IP Controller for Kubernetes
+:version: 1.5
+:type: tutorial
+
+.. _k8s url rewrite:
+
+Rewrite URLs
+============
+
+Overview
+--------
+
+.. include:: /_static/reuse/k8s-version-added-1_5.rst
+
+You can use the |kctlr| to rewrite URLs for Kubernetes Ingress resources and OpenShift Ingresses and Routes.
+
+The Annotations are the same in both orchestration systems. The examples provided in this document are Kubernetes Ingress Resources. Their usage is no different in OpenShift Ingresses or Routes.
+
+.. important::
+
+   If you use both of the virtual-server.f5.com :code:`rewrite` Annotations for the same Service, the Controller applies them in the following order:
+
+   #. :code:`app-root`
+   #. :code:`target-url`
+
+Usage
+-----
+
+You can add the :code:`rewrite` Annotations to any of the following resources:
+
+- Kubernetes Ingress
+- OpenShift Ingress
+- OpenShift Route
+
+.. seealso::
+
+   - :ref:`create k8s ingress`
+   - :ref:`create os route`
+
+When using the Annotation(s) in a Single Service Ingress, Multi-Service Ingress, or Route resource, you can specify a **single value** in the annotation.
+
+When using the Annotation(s) in a Multi-Service Ingress, you can specify a single value --OR-- **multiple, comma-separated** values.
+
+Examples
+--------
+
+Rewrite App-Root
+````````````````
+
+.. sidebar:: Applies to:
+
+   - Kubernetes Single-Service Ingress
+   - Kubernetes Multi-Service Ingress
+   - OpenShift Single-Service Ingress
+   - OpenShift Multi-Service Ingress
+   - OpenShift Route
+
+Use the :code:`rewrite-app-root` Annotation to redirect requests for "/" to a different path.
+
+.. code-block:: YAML
+   :caption: Redirect "/" to "/home" for all traffic
+
+   apiVersion: extensions/v1beta1
+   kind: Ingress
+   metadata:
+     annotations:
+       virtual-server.f5.com/rewrite-app-root: "/home"
+     name: home
+     namespace: default
+   spec:
+     rules:
+     - http:
+       paths:
+       - backend:
+           serviceName: http-svc
+           servicePort: 80
+           path: /home
+
+.. code-block:: YAML
+   :caption: Rewrite "/" to "/en" for traffic with host "docs.example.com"
+
+   apiVersion: extensions/v1beta1
+   kind: Ingress
+   metadata:
+     annotations:
+       virtual-server.f5.com/rewrite-app-root: "docs.example.com=/en"
```

Comment:
docs.example.com=/en -> docs.example.com/docs-en

Reply:
@ryan-talley - What's the reason for this proposed change? This example follows a basic standard for docs websites that use translation (like readthedocs - trying to use real-world examples).

Reply:
The reason here is that the way our controller works is that we create pools and rules to direct to them based on what is in the spec. So if you have docs.example.com=/en as your annotation (and this only applies to app-root) the browser makes a request to docs.example.com/ which then gets a response of 301 redirect to docs.example.com/en the browser then makes another request to docs.example.com/en which will give us a 404 since the BIG-IP only has forwarding rules for docs.example.com/home and docs.example.com/docs-en. So your app-root rewrite path has to match a path in the spec otherwise it won't get forwarded. This does not apply to url-rewrite since it only rewrites the url but does not affect the forwarding decision.

Reply:
@ryan-talley - I didn't realize I had it as docs-en in the path. Fixing that instead of changing the annotation.